src/deluge/deluge.cpp

```diff
@@ -878,59 +907,68 @@ extern "C" int32_t deluge_main(void) {
 #endif
 
 #ifdef TEST_SD_WRITE
+	testSdWrite();
+#endif
+}
 
-	FIL fil; // File object
-	FATFS fs;
-	DIR dp;
-	FRESULT result; //	 FatFs return code
-	int32_t sdTotalBytesWritten;
+extern "C" int32_t deluge_main(void) {
 
-	int32_t count = 0;
+	// HARDWARE SETUP
 
-	while (true) {
+	bool have_oled = checkOLED(); // Detects OLED Screen by DMA settings
 
-		display->setTextAsNumber(count);
+	initPads(have_oled); // Programmable Interrupt Controller (PIC) setup
 
-		int32_t fileNumber = (uint32_t)getNoise() % 10000;
-		int32_t fileSize = (uint32_t)getNoise() % 1000000;
+	initAnalogIO(); // General Purpose (GPIO) pins: analog (line, mic, headphone) ports, battery monitoring, some leds
 
-		char fileName[20];
-		strcpy(fineName, "TEST/") intToString(fileNumber, &fileName[5], 4);
-		strcat(fileName, ".TXT");
+	encoders::init(); // Rotary encoders also handled by GPIO pins
 
-		result = f_open(&fil, fileName, FA_CREATE_ALWAYS | FA_WRITE);
-		if (result) {
-			display->setText("AAAA");
-			while (1) {}
-		}
+	init_OLED_and_CV(have_oled); // OLED and CV share an SPI (serial peripheral interface) channel
 
-		sdTotalBytesWritten = 0;
+	cvEngine.init(have_oled); // CV setup
 
-		while (sdTotalBytesWritten < fileSize) {
-			UINT bytesWritten = 0;
-			result = f_write(&fil, &miscStringBuffer, 256, &bytesWritten);
+	initSPIBSC(); // Set up "serial flash memory". Old comment indicating this also runs audio seems wrong/outdated
 
-			if (bytesWritten != 256) {
-				display->setText("BBBB");
-				while (1) {}
-			}
+	tests_1();
 
-			sdTotalBytesWritten += 256;
-		}
+	// INIT AUDIO AND MIDI
 
-		f_close(&fil);
+	makeParameterRangeConstants(); // Do at compile-time? Constexpr? (not for performance but for tidyness...)
 
-		count++;
-	}
-#endif
+	currentPlaybackMode = &session; // ?
+
+	setOutputState(CODEC.port, CODEC.pin, 1); // Enable Audio Output
+
+	AudioEngine::init();
+
+	audioFileManager.init();
+
+	tests_2();
```

Comment:
Bit of a nit but maybe keep the checks around these? 

Or just remove them, I'm not sure that they actually work anymore

Reply:
Removed the checks for readability. I'm pretty sure the compiler will remove the empty function calls when the defines are negative.

Did not want to remove them in case they were still being used by someone - For now, I added comments to check them for possible future removal...

Reply:
Yeah it will, I think it's more readable to immediately see that they won't run though. Maybe just more descriptive names? 

tests_1 -> automatedActionTests and tests_2 -> hardwareTests

Reply:
Seems that most of theses tests do not even compile when activated, and some seem to hang/crash.

For now I just removed the invocations in deluge_main, 
but left the test code in place, 
as some of it is also called in other ways (e.g. ramTestLED can be started by pressing the three first mod buttons). Also some of the test code might be fixed or at least serve as inspiration for a future test suite?

Reply:
> Seems that most of theses tests do not even compile when activated, and some seem to hang/crash.
> 
> For now I just removed the invocations in deluge_main, but left the test code in place, as some of it is also called in other ways (e.g. ramTestLED can be started by pressing the three first mod buttons). Also some of the test code might be fixed or at least serve as inspiration for a future test suite?

Yeah some of them have already been factored out into the 32bit unit tests suite, it would be nice to move more that way down the road